includes/admin/class-wc-payments-admin.php

```diff
@@ -692,7 +692,7 @@ public function enqueue_payments_scripts() {
 		}
 
 		$screen = get_current_screen();
-		if ( 'shop_order' === $screen->id ) {
+		if ( in_array( $screen->id, [ 'shop_order', 'woocommerce_page_wc-orders' ], true ) ) {
 			$order = wc_get_order();
 
 			if ( WC_Payment_Gateway_WCPay::GATEWAY_ID === $order->get_payment_method() ) {
```

Comment:
This line is now throwing a fatal error on the Order list screen. This is because `$screen->id` equals `woocommerce_page_wc-orders` for both the list and individual order screens when HPOS is enabled.

```
 Fatal error: Uncaught Error: Call to a member function get_payment_method() on bool
```

Perhaps we can check for an order here:
```suggestion
			if ( $order && WC_Payment_Gateway_WCPay::GATEWAY_ID === $order->get_payment_method() ) {
```

Reply:
💯 Good testing. I didn't spot that the orders listing had the same screen id and didn't test.
I was defensive for Multi-Currency but missed the check here. 🙇‍♂️